src/game.cpp

```diff
@@ -13210,11 +13230,11 @@ void game::shift_monsters( const int shiftx, const int shifty, const int shiftz
         if( shiftx != 0 || shifty != 0 ) {
             critter.shift( shiftx, shifty );
         }
-        // TODO: with z-levels, this can be removed, instead shift the critter
-        // along shiftz, too. Than make the 3D-inbounds check.
-        if( shiftz == 0 && m.inbounds( critter.posx(), critter.posy() ) ) {
+
+        if( m.has_zlevels() && m.inbounds( critter.pos() ) ) {
```

Comment:
No it will always despwan if `m.has_zlevels` returns false. I think this condition really needs to include the shiftz parameter.